src/lib/serviceRoutinesV2/postEntity.cpp

```diff
@@ -89,14 +89,24 @@ std::string postEntity
     if ((upcrsP->contextElementResponseVector[ix]->statusCode.code != SccOk) &&
         (upcrsP->contextElementResponseVector[ix]->statusCode.code != SccNone))
     {
+      if (upcrsP->contextElementResponseVector[ix]->statusCode.code == SccInvalidParameter)
+      {
+        // Ugly v1 -> v2 error conversion here :(
+        OrionError error(SccRequestEntityTooLarge, "NoResourcesAvailable", "No more than one geo-location attribute allowed");
```

Comment:
This is related with a similar comment in this PR:

```
    // This v1 -> v2 error conversion is a little crazy (for example, the same
    // SccInvalidParameter at v1 level could match two different error conditions
    // at v2 level, making the logic harder). However, I'm afraid that we need
    // to live with this while NGSIv1 and NGSIv2 coexists :(
```

I will add the same comment also here.

I agree with you: it is dangerous. But empirical evidence (make test) provides relative security.

Hopefully, all this will be "normalized" once NGSIv1 gets deprecated.